src/main/telemetry/smartport.c

```diff
@@ -198,21 +274,30 @@ static void smartPortSendByte(uint8_t c, uint16_t *crcp)
     *crcp = crc;
 }
 
-static void smartPortSendPackage(uint16_t id, uint32_t val)
+static void smartPortSendPackageEx(uint8_t frameId, uint8_t* data)
 {
     uint16_t crc = 0;
-    smartPortSendByte(FSSP_DATA_FRAME, &crc);
-    uint8_t *u8p = (uint8_t*)&id;
-    smartPortSendByte(u8p[0], &crc);
-    smartPortSendByte(u8p[1], &crc);
-    u8p = (uint8_t*)&val;
-    smartPortSendByte(u8p[0], &crc);
-    smartPortSendByte(u8p[1], &crc);
-    smartPortSendByte(u8p[2], &crc);
-    smartPortSendByte(u8p[3], &crc);
+    smartPortSendByte(frameId, &crc);
+    for(unsigned i = 0; i < SMARTPORT_PAYLOAD_SIZE; i++) {
+        smartPortSendByte(*(data++), &crc);
```

Comment:
Well, I found it very confusing. At first I thought "ah, he means do the increment before the data access, ie `*++data`, I had to re-read the code to figure out that you didn't mean this. I'm not against using brackets to improve clarity, but here they obscure things. My thought process was along the lines of "ah brackets, something unusual is happening here. What could it be? Ah yes, do increment before data access. Ah no, he can't possibly mean that..."